streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -5010,14 +5010,23 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
   def repeatZIOChunkOption[R, E, A](
     fa: => ZIO[R, Option[E], Chunk[A]]
   )(implicit trace: Trace): ZStream[R, E, A] =
-    unfoldChunkZIO(fa)(fa =>
-      fa.foldZIO(
-        failure = {
-          case None    => Exit.none
-          case Some(e) => Exit.fail(e)
-        },
-        success = chunk => Exit.succeed(Some((chunk, fa)))
-      )
+    new ZStream(
+      ZChannel.suspend {
+        val fa0 = fa
+
+        def channel: ZChannel[R, Any, Any, Any, E, Chunk[A], Any] =
+          ZChannel.unwrap {
+            fa0.fold(
+              failure = {
+                case None    => ZChannel.unit
+                case Some(e) => ZChannel.fail(e)
```

Comment:
FYI this is more performant:
```scala
case Some(e) => ???
case _ => ???
```

The Scala compiler is overly cautious and if there isn't a wildcard case (i.e., `_`) it will still introduce it in the bytecode which increases the method size and complexity for the C2 compiler to JIT optimize it. Plus, having `case None` means we'd have to still check if `Option(?)` is a `None`, but with `_` we can avoid that